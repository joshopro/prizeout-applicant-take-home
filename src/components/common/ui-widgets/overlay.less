.overlay {
	position: fixed;
	width: 100%;
	height: 100%;
	top: 0;
	left: 0;
	inset: 0;
	background-color: rgba(229, 231, 234, 0.7); // Default - if backdrop-filter isn't supported
	
	// If backdrop-filter is supported
	@supports (backdrop-filter: blur(6px)) {
		background-color: rgba(229, 231, 234, 0.2);
		-webkit-backdrop-filter: blur(6px);
		backdrop-filter: blur(6px);
	}
}
