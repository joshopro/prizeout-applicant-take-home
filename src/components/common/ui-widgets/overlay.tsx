import React from 'react';
import Classnames from 'classnames';
import PropTypes from 'prop-types';

import './overlay.less';

interface OverlayProps {
    onClick?: () => void;
    zIndexType: string;
}

export const Overlay: React.FC<OverlayProps> = ({ zIndexType, onClick }): React.ReactElement => {
    const classes: string = Classnames(`overlay`, { [`z-index-${zIndexType}`]: zIndexType });

    return <div data-testid="overlay" className={classes} onClick={onClick}></div>;
};

Overlay.propTypes = {
    onClick: PropTypes.func,
    zIndexType: PropTypes.string.isRequired,
};
