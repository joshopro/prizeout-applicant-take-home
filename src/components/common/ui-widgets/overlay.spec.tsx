import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { Overlay } from './overlay';

describe('Test Overlay Component', () => {
    const onClick = jest.fn();
    const overlayId = `overlay`;
    const zIndexClass = `z-index-overlay-sidebar`;
    const zIndexType = `overlay-sidebar`;

    // Matches snapshot
    test('Component matches snapshot', () => {
        const { asFragment } = render(<Overlay onClick={onClick} zIndexType={zIndexType} />);

        expect(asFragment()).toMatchSnapshot();
    });

    // Overlay has overlay class
    test('Overlay has overlay class', () => {
        render(<Overlay onClick={onClick} zIndexType={zIndexType} />);

        expect(screen.getByTestId(overlayId)).toHaveClass('overlay');
    });

    // Overlay has sidebar z-index class if sidebar is open and overlay should be below it
    test('Overlay has z-index class', () => {
        render(<Overlay onClick={onClick} zIndexType={zIndexType} />);

        expect(screen.getByTestId(overlayId)).toHaveClass(zIndexClass);
    });
});
