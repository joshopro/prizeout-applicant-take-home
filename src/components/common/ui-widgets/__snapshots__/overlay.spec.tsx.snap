// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Overlay Component Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="overlay z-index-overlay-sidebar"
    data-testid="overlay"
  />
</DocumentFragment>
`;
